astropy/io/fits/fitsrec.py

```diff
@@ -517,6 +521,10 @@ def __getitem__(self, key):
 
         # We got a view; change it back to our class, and add stuff
         out = out.view(type(self))
+        out._heapoffset = self._heapoffset
+        out._heapsize = self._heapsize
+        out._tbsize = self._tbsize
+        out._gap = self._gap
```

Comment:
@mhvk - I never really understood Numpy subclassing but, isn't there a better way to propagate those custom attributes ?